kernel/examples/read-table-single-threaded/src/main.rs

```diff
@@ -114,7 +114,8 @@ fn try_main() -> DeltaResult<()> {
         .build()?;
 
     let mut batches = vec![];
-    for res in scan.execute(engine.as_ref())?.into_iter() {
+    for res in scan.execute(engine.as_ref())? {
+        let res = res?;
```

Comment:
could maybe use a try_for_each? not sure if it's any cleaner though

Reply:
try_for_each only returns results returned in the body, it doesn't fix the issue of getting results in the iterator

Reply:
What about [process_results](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.process_results)?

Reply:
`process_results` ends up having a lot of nesting errors. We get Results in, for which `process_result` outputs `DeltaResult<Output>`, but because we *also* have errors inside the loop, the ouptut itself is an error `DeltaResult<DeltaResult<_>>`. I noticed that we are collecting batches, so I went with a `map_ok`, `flatten_ok`, and `try_collect`.